tensorflow/python/eager/pywrap_tfe_src.cc

```diff
@@ -3033,6 +3048,11 @@ PyObject* RecordGradient(PyObject* op_name, PyObject* inputs, PyObject* attrs,
 
   string c_op_name = TFE_GetPythonString(op_name);
 
+  PyObject* device = DeviceFromTensorSeq(results);
+  if (device == Py_None) {
+    device = DeviceFromTensorSeq(inputs);
```

Comment:
Maybe I'm missing something, we have a reference to "device" here. It's incremented when we create a backward function and decremented when that function is deleted. But do we ever get rid of that base reference?

Reply:
You are right, I pushed a fixed version.